pkg/storage/etcd/etcd_helper.go

```diff
@@ -289,26 +290,76 @@ func (h *etcdHelper) Set(ctx context.Context, key string, obj, out runtime.Objec
 	return err
 }
 
+func checkPreconditionsMatch(preconditions *api.Preconditions, out runtime.Object) error {
+	if preconditions == nil {
+		return nil
+	}
+	objMeta, err := api.ObjectMetaFor(out)
+	if err != nil {
+		return kerrors.NewInternalError(err)
```

Comment:
Something to the effect of "can't enforce preconditions %v on un-introspectable object type %v". Need to interpret and produce NewInternalError in the deletion error interpretation function.